scripts/requirements-doc.txt

```diff
@@ -1,8 +1,8 @@
 # DOC: used to generate docs
 
-breathe>=4.9.1,<4.15.0
-docutils>=0.14
-sphinx>=1.7.5,<3.0
+breathe>=4.20.0
+docutils>=0.16
+sphinx>=3.2.0
```

Comment:
Only requiring >=3.2.0 for sphinx seems dangerous to me. In the past sphinx changed a lot in their api that broke compatibility with extensions and themes when updating major versions.

Reply:
What is your suggestion?

Reply:
Maybe `sphinx>=3.2.0,<4.0` would be good for now.

Reply:
Ok, the latest release is `3.2.1` and there are no development releases with updated major versions, so it's really hard to know right now if they'll be compatible or not, but I will update as suggested.

Reply:
I do not think this is an issue. If `sphinx` follows semantic versioning (https://semver.org/) then change in major version is exactly doing what might have been expected.

Reply:
I have misunderstood the problem. Sorry.

Reply:
I changed it to what @chrta suggested; my personal opinion would be to not set upper limits right now, but they won't really affect us for a few months at least.